daemon/entrypoint.sh

```diff
@@ -28,6 +32,89 @@ if [[ ! -e /etc/ceph/${CLUSTER}.conf ]]; then
 fi
 }
 
+function create_mon_ceph_config_from_kv {
+
+  CLUSTER_PATH=ceph-config/${CLUSTER}
+
+  echo "Adding Mon Host - ${MON_NAME}"
+  kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} cas mon_host/${MON_NAME} ${MON_IP} > /dev/null 2>&1
+
+  # Acquire lock to not run into race conditions with parallel bootstraps
+  until kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} cas ${CLUSTER_PATH}/lock $MON_NAME > /dev/null 2>&1 ; do
+    echo "Configuration is locked by another host. Waiting."
+    sleep 1
+  done
+
+  # Update config after initial mon creation
+  if kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} get ${CLUSTER_PATH}/monSetupComplete > /dev/null 2>&1 ; then
+    echo "Configuration found for cluster ${CLUSTER}. Writing to disk."
+
+
+    until ./confd -onetime -backend ${KV_TYPE} -node ${KV_IP}:${KV_PORT} -prefix="/${CLUSTER_PATH}/" -confdir="./conf.d" ; do
+      echo "Waiting for confd to update templates..."
+      sleep 1
+    done
+
+    echo "Adding Keyrings"
+    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} get ${CLUSTER_PATH}/monKeyring > /etc/ceph/ceph.mon.keyring
+    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} get ${CLUSTER_PATH}/adminKeyring > /etc/ceph/ceph.client.admin.keyring
+
+    if [ ! -f /etc/ceph/monmap ]; then
```

Comment:
That monSetupComplete condition is still run after initial set up on the same host to keep the ceph.conf up to date so there are cases where monmap can exist. The check for the monmap was to avoid overwriting if the monmap has been modified (new hosts added, etc).

I probably should spend a little more time getting familiar with the ceph monmap code since I'm not entirely clear on what happens in cases of old monmaps or new hosts with changing hostnames ;)